battleship-client/src/main/java/com/academy/solid/nie/client/ui/Board.java

```diff
@@ -142,7 +142,8 @@ private void markEndOfShip() {
      * @param ship - represents single instance of Ship
      */
     private boolean canPlaceShip(final Ship ship) {
-        return ship.getPositions().stream().noneMatch(point2D -> !isInScope(point2D) || boardFX.isOccupied(point2D) || isThereShipInTheNeighborhood(point2D));
+        return ship.getPositions().stream().noneMatch(point2D ->
```

Comment:
The code is pretty and I like the streaming. Problem - you are mixing your game logic here WITH JavaFX code. I'd advise splitting this more decisively, but won't block a PR over it.

I'm not addressing just this line or this method, I'm rather speaking about whole Board class.